persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/query/InfinispanQuery.java

```diff
@@ -145,6 +145,9 @@ protected InfinispanQuery(QueryFactory qf, String rootType) {
     }
 
     private String getRecursiveString(AttributeFilter<?> filter, String joining) {
-        return ((List<AttributeFilter<?>>) filter.getValue()).stream().map(filterStringFunction()).collect(joining(joining));
+        return "(" + ((List<AttributeFilter<?>>) filter.getValue())
+                .stream()
+                .map(filterStringFunction())
+                .collect(joining(joining)) + ")";
```

Comment:
perhaps change to, that way you can remove the outer strings.
```suggestion
                .collect(joining(joining, "(", ")")) + ")";
```

Reply:
good catch @cristianonicolai